### DISCUSSION TITLE
Pass master url with http basic auth

### DISCUSSION THREAD

**Discussion by cypreess** [devpi/devpi#993] [Q&A]
Hi,

I would like to use devpi as my localhost mirror of external private pypi index (that just happenes to also by a devpi). To achieve that I must authorize in external pypi index therefore I wanted to pass `--master-url http://<user>:<pass>@devpi.someplace...` 

devpi does not work as expected. Calls to master index does not have basic auth being passed (even though I have checked in the code, it seems they are being parsed out from the URL).

Any help here?

**Comment by fschulze** [comment]
The replication ``/+changelog`` endpoint already uses authentication via the shared secret, so the basic authentication should be disabled for it. If you don't want replication, but an actual mirror index, then the ``mirror_url`` option does support basic authentication.

**Reply by fschulze** [reply]
Basic authentication in general is discouraged and support for in in ``devpi-client`` is deprecated. You should look into ``devpi-tokens`` as a replacement.
